### ISSUE TITLE
seems cool but... lol

### ISSUE THREAD

**Issue by DrBlackross** [jefc1111/zenbot-sim-runner#5]
"in a nutshell, you need to install the dependencies, tell it where your working instance of Zenbot is, import some data from Zenbot and away you go!"

(i think i found that part)

the but is... doesn't zenbot use mongodb? not mysql

**Comment by DrBlackross** [comment]
(trust me i went through this years ago tweeking zenbot and gecko in node)

**Comment by DrBlackross** [comment]
i keep getting stuck at the 'composer install' step

~/zenbot-sim-runner$ composer install
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. It is recommended that you run `composer update` or `composer update <package name>`.
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for tijsverkoyen/css-to-inline-styles 2.2.3 -> satisfiable by tijsverkoyen/css-to-inline-styles[2.2.3].
    - tijsverkoyen/css-to-inline-styles 2.2.3 requires ext-dom * -> the requested PHP extension dom is missing from your system.
  Problem 2
    - Installation request for phar-io/manifest 2.0.1 -> satisfiable by phar-io/manifest[2.0.1].
    - phar-io/manifest 2.0.1 requires ext-dom * -> the requested PHP extension dom is missing from your system.
  Problem 3
    - Installation request for phpunit/php-code-coverage 9.2.5 -> satisfiable by phpunit/php-code-coverage[9.2.5].
    - phpunit/php-code-coverage 9.2.5 requires ext-dom * -> the requested PHP extension dom is missing from your system.
  Problem 4
    - Installation request for phpunit/phpunit 9.5.2 -> satisfiable by phpunit/phpunit[9.5.2].
    - phpunit/phpunit 9.5.2 requires ext-dom * -> the requested PHP extension dom is missing from your system.
  Problem 5
    - Installation request for theseer/tokenizer 1.2.0 -> satisfiable by theseer/tokenizer[1.2.0].
    - theseer/tokenizer 1.2.0 requires ext-dom * -> the requested PHP extension dom is missing from your system.
  Problem 6
    - tijsverkoyen/css-to-inline-styles 2.2.3 requires ext-dom * -> the requested PHP extension dom is missing from your system.
    - laravel/framework v8.28.1 requires tijsverkoyen/css-to-inline-styles ^2.2.2 -> satisfiable by tijsverkoyen/css-to-inline-styles[2.2.3].
    - Installation request for laravel/framework v8.28.1 -> satisfiable by laravel/framework[v8.28.1].

  To enable extensions, verify that they are enabled in your .ini files:
    - /etc/php/7.4/cli/php.ini
    - /etc/php/7.4/cli/conf.d/10-opcache.ini
    - /etc/php/7.4/cli/conf.d/10-pdo.ini
    - /etc/php/7.4/cli/conf.d/20-calendar.ini
    - /etc/php/7.4/cli/conf.d/20-ctype.ini
    - /etc/php/7.4/cli/conf.d/20-exif.ini
    - /etc/php/7.4/cli/conf.d/20-ffi.ini
    - /etc/php/7.4/cli/conf.d/20-fileinfo.ini
    - /etc/php/7.4/cli/conf.d/20-ftp.ini
    - /etc/php/7.4/cli/conf.d/20-gettext.ini
    - /etc/php/7.4/cli/conf.d/20-iconv.ini
    - /etc/php/7.4/cli/conf.d/20-json.ini
    - /etc/php/7.4/cli/conf.d/20-mbstring.ini
    - /etc/php/7.4/cli/conf.d/20-phar.ini
    - /etc/php/7.4/cli/conf.d/20-posix.ini
    - /etc/php/7.4/cli/conf.d/20-readline.ini
    - /etc/php/7.4/cli/conf.d/20-shmop.ini
    - /etc/php/7.4/cli/conf.d/20-sockets.ini
    - /etc/php/7.4/cli/conf.d/20-sysvmsg.ini
    - /etc/php/7.4/cli/conf.d/20-sysvsem.ini
    - /etc/php/7.4/cli/conf.d/20-sysvshm.ini
    - /etc/php/7.4/cli/conf.d/20-tokenizer.ini
  You can also run `php --ini` inside terminal to see which files are used by PHP in CLI mode.

php --version
PHP 7.4.3 (cli) (built: Jul  5 2021 15:13:35) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.3, Copyright (c), by Zend Technologies

**Comment by DrBlackross** [comment]
OK, ran

sudo apt install php-xml

fixed  composer install issue

**Comment by DrBlackross** [comment]
ok going to install mysql (i think)

~/zenbot-sim-runner$ php artisan migrate
**************************************
*     Application In Production!     *
**************************************

 Do you really wish to run this command? (yes/no) [no]:
 > yes


   Illuminate\Database\QueryException 

  could not find driver (SQL: select * from information_schema.tables where table_schema = forge and table_name = migrations and table_type = 'BASE TABLE')

  at vendor/laravel/framework/src/Illuminate/Database/Connection.php:678
    674▕         // If an exception occurs when attempting to run a query, we'll format the error
    675▕         // message to include the bindings with SQL, which will make this exception a
    676▕         // lot more helpful to the developer instead of just the database's errors.
    677▕         catch (Exception $e) {
  ➜ 678▕             throw new QueryException(
    679▕                 $query, $this->prepareBindings($bindings), $e
    680▕             );
    681▕         }
    682▕ 

      +33 vendor frames 
  34  artisan:37
      Illuminate\Foundation\Console\Kernel::handle()





using this guide 

https://www.digitalocean.com/community/tutorials/how-to-install-mysql-on-ubuntu-20-04

**Comment by DrBlackross** [comment]
alright, using this guide... lol

https://stackoverflow.com/questions/46745365/artisan-migrate-could-not-find-driver

~/zenbot-sim-runner$ sudo find / -name php.ini
/home/drblackross/zenbot-sim-runner/vendor/laravel/sail/runtimes/8.0/php.ini
/home/drblackross/zenbot-sim-runner/vendor/laravel/sail/runtimes/7.4/php.ini
/etc/php/7.4/cli/php.ini

either way it didn't work lol 
its a cool idea, but forge wasn't able to access the sql database in the end and add the tables, even though i did add the user lol... i can wait till some of the bugs get worked out.

if it can work with mongodb that'll be cool

(phpnuke wasnt this hard lol)

**Comment by jefc1111** [comment]
Yeah, getting everything to hang together working properly is quite an 'expert' undertaking really, and I just haven't had the time to refine the install process. Thanks for persevering!

However ... I have been working on a Docker-based deployment which includes all the many dependencies (even Zenbot and MongoDB). I've got it deployed in the cloud I just need to set up a domain name of some sort. Once that is sorted I will let you know in this thread and you can sign up for a login. I wish I could get it done right now but have a lot of other responsibilities. I'd hope to be able to get back to you within a few days though.
